### PR TITLE
Reporting: Fix the `View report` link URL on Total payment volume tile, within Payment activity widget

### LINKED ISSUE #8706: [Task] - Fix the `View report` link on the Total payment volume tile
The `View report` link on the Total payment volume tile should lead to the `Transactions` page showing all types created over the last 7 days (or selected time filter) except for `Loan disbursement` and `Loan repayment`

It is currently showing all types created over the last 7 days (or selected time filter) . We need to make sure `Loan disbursement` and `Loan repayment` are excluded. 

This may require creating an additional advanced filter on the Transactions page, to accommodate the above rule 

#### Context
P2 - pdjTHR-3DQ-p2#comment-4954

```[tasklist]
### Subtasks
- [ ] #8726
- [ ] https://github.com/Automattic/woocommerce-payments-server/issues/5247
```

### PR DESCRIPTION
**Pull request by nagpai**
Fixes #8706 

#### Changes proposed in this Pull Request

The PR fixes the `View report` link on Total Payment volume tile to match the requirement as follows: 

* `Total payment volume` should show all types created over the last 7 days (or selected time filter) except for `Loan disbursement` and `Loan repayment`

#### Notes

* The existing advanced filter on the Transaction page, do have `Type is` and `Type is not` filters, but they do not allow selecting more than one types in it, in the current design. 
* Thanks to @shendy-a8c , who drew my attention to the default Disputes link that uses search term params to give a result comprising types - `needs_response` and `warning_needs_response` 
image

* For segregating transactions matching the above criteria, similar principle has been used :
  * On the server, the [exact search](https://github.com/Automattic/woocommerce-payments-server/blob/b053dfccfe5ba8bd1545c71394a501bcffce02f1/server/wp-content/rest-api-plugins/endpoints/wcpay/repository/class-transaction-repository.php#L43-L47) of `type` has been added in the search fields. 
  * On the client, the `View report` URL was edited to include the search terms. 
  * Some minor CSS changes were done so that the tags that appear in the search box do not warp outside the box. 

A more permanent solution to this would be to have `typeIs` and `typeIsNot` params accept arrays. But this may require higher effort to execute and to ensure backward compatibility. I found using the way similar to Disputes a better option for now.

#### Testing instructions
* On your local server instance, please check out branch `add/type-to-search-by-fields`
* On the local test client setup, make sure your account has sufficient transactions within the last 7 days, and at least one `Loan disbursement` and `Loan repayment` .  See if you can switch to `acct_1OuYn1QpUShfetUA` which has these ( and should remain for the next few days. Else create the loan transactions using the steps in the PdjTHR-31T-p2. 
* Browse to `Payments > Overview` and click the `View report` against `Total payment volume`
* On the transactions report page that lands, you should see all transactions except Loan disbursement and Loan repayment

#### Screenshots

On a wider screen
image

On a narrow screen
image

 Without the CSS change, it would appear like this:
image

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8726` or branch name `fix/8706-tpv-view-report-link` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8706-tpv-view-report-link) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8cd9331bd9ff32992e53640a3b0a41f8d5fc4cf8
- Build time: 2024-05-02 08:54:27 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/fix-8706-tpv-view-report-link [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: fix
+Comment: Fix the `View report` link on Total payment volume tile on the Payment activity widget. Changes behind feature flag, and part of larger change that adds the wPayment activity widget.
```
Comment (jessy-p) [inline changelog/fix-8706-tpv-view-report-link]:
```suggestion
Comment: Fix the `View report` link on Total payment volume tile on the Payment activity widget. Changes behind feature flag, and part of larger change that adds the Payment activity widget.
```

Comment (nagpai) [reply changelog/fix-8706-tpv-view-report-link]: Sorry about that typo. Thanks for catching it! - 7930735

**Comment by jessy-p** [comment]
The transaction filter is visible on the page, however if I try to apply filtering by a type of transaction, it does not work. This will be confusing for the customer.

**Review by jessy-p — COMMENTED** [review]
Filtered values are NOT displayed on the page. Hope that is OK/ as expected @nagpai 
Further filtering by applying `Advanced` filter is working as expected.

**Comment by jessy-p** [comment]
> Some minor CSS changes were done so that the tags that appear in the search box do not warp outside the box.

If I do not have `Loan disbursement` and `Loan repayment` types in my timeline, should I still be seeing all the other transaction types as tags? Can you share a screenshot? @nagpai

**Comment by nagpai** [comment]
>Filtered values are NOT displayed on the page. Hope that is OK/ as expected @nagpai
Further filtering by applying Advanced filter is working as expected.

Hi @jessy-p , I hope you are seeing all transaction types except for `Loan disbursement` and `Loan repayment` 

>If I do not have Loan disbursement and Loan repayment types in my timeline, should I still be seeing all the other transaction types as tags? Can you share a screenshot?

Yes, it is expected for you to see all other transaction type tags in the search bar, since we do not have a negative search option .. The types include `charge, payment, payment_failure_refund, payment_refund, refund, refund_failure, dispute, dispute_reversal, card_reader_fee` .

Since this list is long, to prevent warping I added a subtle scroll overflow.  ~Adding the screenshot in a moment.~  Added! 

If you like to see the loan related transactions filtered out, feel free to switch to my test account via the CLI command on server - `./local/bin/link-account.sh <your client blog id>  acct_1OuYn1QpUShfetUA` . It has a loan disbursal and repayment transaction, that you can see getting filtered out.

**Comment by jessy-p** [comment]
After using the server branch `add/type-to-search-by-fields`, I am able to see the Transactions search box with the tags as expected.

🟡  Only if the customer clicks on `Advanced` filter, he will see the dates which have been applied. Add `filter=advanced` to the URL to have the Date displayed when the page is loaded.

✅  Filters are working correctly and when changed, the data refreshes accordingly. Everything clear, nothing confusing or odd observed.

🟡  The scroll section still looks bit off to me. Bottom part of the tags seem to be cut off  and when you click in the box, the `Search by` prompt overlaps with `x`. Can check with Design about solutions for this.
image

**Comment by nagpai** [comment]
>🟡 Only if the customer clicks on Advanced filter, he will see the dates which have been applied. Add filter=advanced to the URL to have the Date displayed when the page is loaded.

I have addressed this here 75a17d2

It looks like this now with the advanced date filters visible.

image

**Comment by nagpai** [comment]
@jessy-p  

On this point 

>🟡 The scroll section still looks bit off to me. Bottom part of the tags seem to be cut off and when you click in the box, the Search by prompt overlaps with x. Can check with Design about solutions for this.

I tried on Chrome, Vivaldi, FF and Safari at my end, and I am unable to replicate the bottom clipping on the search box tags. The `Search by prompt` does overlap with X for certain in-betweeen screen sizes. 

In its current state, the landing link shows the Total Payment Volume numbers and its breakdown. But it makes further filtration by the user a bit uncomfortable. The recording here may illustrate this better. 

https://github.com/Automattic/woocommerce-payments/assets/4162931/56e490c3-e2d9-425e-9d85-545995a24af1

@rogermattic , Hi, we need your help here for the view of the search bar within Transactions report results - when 
* There are a larger number of search terms, probably warping into next line
* More likely on a mobile or a narrow screen profile

 In our case because of the filter needed for TPV transactions ). This disrupts the appearance of the search prompt. It is possible to add further search tags though, but visibility is a bit bad.

@jessy-p  I was wondering if I should create a separate case for this. Would this be a blocker for the current PR?

File: client/transactions/list/style.scss [hunk 2]
```diff
@@ -123,6 +123,8 @@ $gap-small: 12px;
 
 			.woocommerce-select-control__control {
 				height: 38px;
+				overflow: scroll;
+				align-items: flex-start;
```
Comment (Jinksi) [inline client/transactions/list/style.scss]: Although it is a minor fix, this change isn't behind the feature flag.

Perhaps it would help communicate this change if we had 2 changelog entries:
1. Fix for view report behind feature flag.
2. Fix for Transaction List styles


Alternatively, relevant to this comment:

> @jessy-p I was wondering if I should create a separate case for this. Would this be a blocker for the current PR?

We could move this change to a separate issue/PR. This may help us step back and look at the problem/solution separate to the Payment Activity Card.

Comment (Jinksi) [reply client/transactions/list/style.scss]: As an alternative UX to `overflow: scroll`, I would consider letting the search box to increase in height when necessary, example:

image

image

_(avoiding the `!important` if possible 😉)_

Comment (nagpai) [reply client/transactions/list/style.scss]: >Although it is a minor fix, this change isn't behind the feature flag.

I had completely missed this! Thanks for sharing!. 

~I would favor creating a separate issue + PR. It would be tiny but keep things clear.~

I did a deeper check and it looks like we need some more work to do for responsive view ( for narrow screens ) . e.g. The Date filter with `Between` appears botched in mobile screen, and the search prompt disappears too in the narrow screen. I will create a separate issue for this. 

Within this PR however, I will include the change @Jinksi  suggested to increase the height. - 75a65a4. Since it will help immediately with how the tags appear when there are too many.

**Comment by nagpai** [comment]
Via - 16d12d2

Moved the view report search params generating function outside, to help add similar terms to other tiles 

```typescript
const searchTermsForViewReportLink = {
		totalPaymentVolume: [
			'charge',
			'payment',
			'payment_failure_refund',
			'payment_refund',
			'refund',
			'refund_failure',
			'dispute',
			'dispute_reversal',
			'card_reader_fee',
		],
	};

	const getSearchParams = ( searchTerms: string[] ) => {
		return searchTerms.reduce(
			( acc, term, index ) => ( {
				...acc,
				[ `search[${ index }]` ]: term,
			} ),
			{}
		);
	};
``` 

For refunds, charges and disputes, there will be similar arrays added like `totalPaymentVolume` seen above. 

and the `getSearchParams` function will add each search term in the URL.

File: client/components/payment-activity/payment-activity-data.tsx [hunk 3]
```diff
@@ -43,6 +43,30 @@ const PaymentActivityData: React.FC = () => {
...
+			'payment_refund',
+			'refund',
+			'refund_failure',
+			'dispute',
+			'dispute_reversal',
+			'card_reader_fee',
+		],
+	};
+
+	const getSearchParams = ( searchTerms: string[] ) => {
```
Comment (Jinksi) [inline client/components/payment-activity/payment-activity-data.tsx]: 💭 This is not an issue per se (feel free to ignore), but I'd consider moving this function and `searchTerms` declaration outside of the React component scope (alongside `getDateRange`) since it doesn't require any component variables/state.

E.g. if it is declared outside the component, it explicitly lets us know it is decoupled from the component and can be used elsewhere, moved, unit tested, etc.

**Review by Jinksi — APPROVED** [review]
This tests well @nagpai 🙌

I've tested with the account mentioned in the description and the view report link works as expected.

On iOS, the search input height may not be visually ideal, but it works well.

As you suggest, I agree we should followup on the mobile UI issues with the transaction filters as separate issues.

File: client/components/payment-activity/payment-activity-data.tsx (hunk 3 above)
Comment (nagpai) [reply client/components/payment-activity/payment-activity-data.tsx]: That is a valuable feedback. Note-to-self on deciding where to scope. 🙌 .

Comment (nagpai) [reply client/components/payment-activity/payment-activity-data.tsx]: done here!  

https://github.com/Automattic/woocommerce-payments/pull/8726/commits/4b53bac09d25f557600598492c074810f0c8a5fa

File: client/transactions/list/style.scss (hunk 2 above)
Comment (rogermattic) [reply client/transactions/list/style.scss]: @nagpai Thanks for rising it and @Jinksi thanks for coming up with the fix. I feel like the increase in height is the best we can do now. For the narrower (mobile) screens - I don't know if this is possible – but I'd:
1. put the applied filter (tags) in one line (could they be horizontally scrollable?) 
2. have the `x` cross sign for the user to clear the filters visible 
2.  have the next (2nd) line empty so that the user can search if they want

In a nutshell, I'd prioritise the user's ability to search over seeing all the applied filters.

Ugh, does it make sense?

Comment (nagpai) [reply client/transactions/list/style.scss]: @rogermattic  Thanks, Magda! I will add this guidance in the new issue I create for improving the mobile UI 🙏 !

Comment (rogermattic) [reply client/transactions/list/style.scss]: Thanks @nagpai ! if you have anything for me to look at, just ping me:)

Comment (nagpai) [reply client/transactions/list/style.scss]: I have created https://github.com/Automattic/woocommerce-payments/issues/8781 to track this change here!
